code/modules/projectiles/ammunition/magazines.dm

```diff
@@ -258,6 +258,51 @@
 	caliber = "4.6x30mm"
 	max_ammo = 20
 	multi_sprite_step = 4
+	multiload = 0
+	w_class = WEIGHT_CLASS_NORMAL
+	///A var to check if the mag is being loaded
+	var/being_loaded = FALSE
+	/// There are two reloading processes ongoing so cancel them
+	var/double_loaded = FALSE
+
+/obj/item/ammo_box/magazine/wt550m9/attackby(obj/item/A, mob/user, params)
+	if(istype(A, /obj/item/ammo_box/wt550) || istype(A, /obj/item/ammo_box/magazine/wt550m9))
+		to_chat(user, "<span class='notice'>You begin to load the magazine with [src].</span>")
+		var/obj/item/ammo_box/AB = A
+		if(being_loaded == FALSE && double_loaded != TRUE)
+			being_loaded = TRUE
+			for(var/obj/item/ammo_casing/AC in AB.stored_ammo)
+				if(double_loaded == FALSE)// checks to make sure double_loaded hasn't been made true while an ongoing reload is already ongoing
+					if(length(stored_ammo) < max_ammo)
+						if(moving_do_after(user, 0.5 SECONDS, target = src, use_default_checks = TRUE))
```

Comment:
I feel like a `do_after_once` would help here reduce some of the checks.

Reply:
Yeah, I'm not keen on the `double_loaded` var.